morphir/sdk/core/src/morphir/sdk/Dict.scala

```diff
@@ -19,16 +19,36 @@ package morphir.sdk
 import morphir.sdk.Maybe._
 
 object Dict {
-  sealed abstract class Dict[K, +V]
-  private case object EmptyDict                              extends Dict[Any, Nothing]
-  private case class DictImpl[K, +V](val wrapped: Map[K, V]) extends Dict[K, V]
+  sealed abstract class Dict[-K, +V]
+  private case object EmptyDict                          extends Dict[Any, Nothing]
+  private case class DictImpl[K, +V](wrapped: Map[K, V]) extends Dict[K, V]
 
-  def empty[K, V]: Dict[K, V] = EmptyDict.asInstanceOf[Dict[K, V]]
+  /* Build */
+  def empty[K, V]: Dict[K, V] = EmptyDict
 
-  def get[K, V](targetKey: K)(dict: Dict[K, V]): Maybe[V] =
+  def singleton[K, V](key: K)(value: V): Dict[K, V] = DictImpl(Map(key -> value))
+
+  def insert[K, V](key: K)(value: V)(dict: Dict[K, V]): Dict[K, V] =
     dict match {
-      case EmptyDict         => Maybe.Nothing
-      case DictImpl(wrapped) => wrapped.get(targetKey)
+      case EmptyDict         => DictImpl(Map(key -> value))
+      case DictImpl(wrapped) => DictImpl(wrapped + (key -> value))
+    }
+
+  def update[K, V](targetKey: K)(updatedValue: V)(dict: Dict[K, V]): Dict[K, V] = dict match {
+    case EmptyDict         => dict
+    case DictImpl(wrapped) => DictImpl(wrapped.updated(targetKey, updatedValue))
+  }
+
+  def remove[K, V](targetKey: K)(dict: Dict[K, V]): Dict[K, V] = dict match {
+    case EmptyDict         => dict
+    case DictImpl(wrapped) => DictImpl(wrapped.filterKeys(key => key != targetKey))
```

Comment:
The dictionary might become empty after the filtering so we need an additional check here and return EmptyDict in that case.